docs/src/pages/customization/dark-mode/dark-mode.md

```diff
@@ -0,0 +1,132 @@
+# Dark mode
+
+<p class="description">Material-UI comes with two palette modes: light (the default) and dark.</p>
+
+You can make the theme dark by setting `mode: 'dark'`.
+
+```js
+const darkTheme = createTheme({
+  palette: {
+    mode: 'dark',
+  },
+});
+```
+
+While it's only a single value change, the `createTheme` helper modifies several palette values.
+The colors modified by the palette mode are the following:
+
+{{"demo": "pages/customization/dark-mode/DarkTheme.js", "bg": "inline", "hideToolbar": true}}
+
+> Note: The colors are modified only if you use the default palette. If you have a custom palette, you need to make sure that you have the correct values based on the `mode`. The following section explains how you can do it.
+
+## Dark mode with custom palette
+
+The easiest way of how you can implement your custom palette that depends on mode is to have a dedicated function that will return the palette based on the mode. For example:
+
+```ts
+const getDesignTokens = (mode: PaletteMode) => ({
+  palette: {
+    mode,
+    ...(mode === 'light'
+      ? {
+          // palette values for light mode
+          primary: amber,
+          divider: amber[200],
+          text: {
+            primary: grey[900],
+            secondary: grey[800],
+          },
+        }
+      : {
+          // palette values for dark mode
+          primary: deepOrange,
+          divider: deepOrange[700],
+          background: {
+            default: deepOrange[900],
+            paper: deepOrange[900],
+          },
+          text: {
+            primary: '#fff',
+            secondary: grey[500],
+          },
+        }),
+  },
+});
+```
+
+You can see on the example that there are different colors used based on whether the mode is light or dark. The next step is to use this function when creating the theme.
+
+```tsx
+export default function App() {
+  const [mode, setMode] = React.useState<PaletteMode>('light');
+  const colorMode = React.useMemo(
+    () => ({
+      // The dark mode switch would invoke this method
+      toggleColorMode: () => {
+        setMode((prevMode: PaletteMode) =>
+          prevMode === 'light' ? 'dark' : 'light',
+        );
+      },
+    }),
+    [],
+  );
+
+  // Update the theme only if the mode changes
+  const theme = React.useMemo(() => createTheme(getDesignTokens(mode)), [mode]);
+
+  return (
+    <ColorModeContext.Provider value={colorMode}>
+      <ThemeProvider theme={theme}>
+        <Page />
+      </ThemeProvider>
+    </ColorModeContext.Provider>
+  );
+}
+```
+
+Here is a fully working example:
+
+{{"demo": "pages/customization/dark-mode/DarkThemeWithCustomPalette.js", "defaultCodeOpen": false}}
+
+## Toggling color mode
```

Comment:
The previous section already does this. Maybe we should introduce it before?

But then `DarkThemeWithCustomPalette` feel almost like `ToggleColorMode`.

Maybe it should be only one section, and about toggling the color mode.

Reply:
Actually in `DarkThemeWithCustomPalette` would removing the context simplify the demo?

Reply:
Removed the context.